editor/modes/visual-editor/block.js

```diff
@@ -251,12 +314,54 @@ class VisualEditorBlock extends Component {
 
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
+		const { previousBlock, nextBlock, onFocus } = this.props;
+		const up = keyCode === UP;
+		const down = keyCode === DOWN;
+		const left = keyCode === LEFT;
+		const right = keyCode === RIGHT;
+
+		if ( up || down || left || right ) {
+			const editor = target.id && tinymce.get( target.id );
+
+			if ( target === this.node ) {
+				const reverse = up || left;
+				const followingBlock = reverse ? previousBlock : nextBlock;
+
+				if ( followingBlock ) {
+					event.preventDefault();
+					onFocus( followingBlock.uid, { offset: reverse ? -1 : 0 } );
+				}
+			}
+
+			if ( editor ) {
+				const reverse = up || left;
+				const followingBlock = reverse ? previousBlock : nextBlock;
+
+				if ( ! isEdge( { editor, reverse } ) ) {
+					return;
```

Comment:
A `return` here makes me wary since we don't isolate this logic from other `onKeyDown` handlers. Currently the only remaining logic which would be skipped is the `ENTER` handling, that would never coincide with this, but seems easy to overlook in future maintenance.